lib/Types.php

```diff
@@ -30,6 +30,20 @@
  *  prefix?: string,
  *  delegate: array<string>
  * }
+ * @phpstan-type RelationOptions array{
```

Comment:
Don't repeat yourself. Have this live in Relation::$VALID_OPTIONS or Type::$VALID_OPTIONS, one or the other. The Relation::$VALID_OPTIONS does not have mapped_names. Should it? I would also list these items in alphabetical order.

Reply:
Absolutely! I looked into it a bit, and all the methods and functionality around that structure seemed to be fairly pointless, so I tore it all out.

bb4c294